From c5f8237bba87dcf113e41cd53a723ce1dad2c3dd Mon Sep 17 00:00:00 2001
From: Laurent Monin <github@norz.org>
Date: Mon, 20 Aug 2018 14:08:11 +0200
Subject: [PATCH] PICARD-1318: Fix dictionary changed size during iteration
 error

It happens when looking up a track with script $set(newartists, %artists%)
---
 picard/metadata.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/picard/metadata.py b/picard/metadata.py
index 8e2aa5266..4f9269b0c 100644
--- a/picard/metadata.py
+++ b/picard/metadata.py
@@ -333,7 +333,7 @@ def rawitems(self):
     def apply_func(self, func):
         for key, values in self.rawitems():
             if key not in PRESERVED_TAGS:
-                self[key] = [func(value) for value in values]
+                super().__setitem__(key, [func(value) for value in values])
 
     def strip_whitespace(self):
         """Strip leading/trailing whitespace.
